pkg/sql/pg_catalog.go

```diff
@@ -3559,6 +3623,18 @@ func (h oidHasher) rewriteOid(source descpb.ID, depended descpb.ID) *tree.DOid {
 	return h.getOid()
 }
 
+// DBSchemaRoleOid creates an OID based on the combination of a db/schema/role.
+// This is used to generate a unique row identifier for pg_default_acl.
+func (h oidHasher) DBSchemaRoleOid(
+	dbID descpb.ID, scName string, normalizedRole string,
+) *tree.DOid {
+	h.writeTypeTag(dbSchemaRoleTypeTag)
+	h.writeDB(dbID)
+	h.writeSchema(scName)
```

Comment:
could we use schemaID here instead of name?

Reply:
It's a current design of the OidHasher - schema names are used instead. Don't think we should change this as it'll change existing oids.

Ie `pg_namespace` already uses the schema name to generate the oid.